gpii/node_modules/gpii-iniFile/index.js

```diff
@@ -0,0 +1,454 @@
+/*
+ * INI File reader/writer
+ *
+ * Copyright 2018 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The R&D leading to these results received funding from the
+ * Department of Education - Grant H421A150005 (GPII-APCP). However,
+ * these results do not necessarily represent the policy of the
+ * Department of Education, and you should not assume endorsement by the
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+var fs = require("fs"),
+    os = require("os"),
+    XRegExp = require("xregexp");
+
+gpii.iniFile = fluid.registerNamespace("gpii.iniFile");
+
+var patten = fs.readFileSync(__dirname  + "/ini.regex");
+gpii.iniFile.regex = new XRegExp(patten, "gmsx");
+
+/**
+ * Parse an INI file, invoking call backs at interesting points: start/end of a section, and on a key=value.
+ *
+ * If options.write=true, the return values of the callbacks will be used to modify the content, the entirety of which
+ * is then returned by this function.
+ *
+ * It uses the regular expression in ./ini.regex to perform the actual parsing.
+ *
+ * @param {String} content The INI file content.
+ * @param {Object} handlers Object containing the callbacks.
+ * @param {Function} handlers.sectionBegin(sectionPath) - called at the start of a new section, returning a string to
```

Comment:
Use http://usejsdoc.org/tags-callback.html to document the arguments and return of this function

Reply:
One step closer to strong types